WeakAuras/SubscribableObject.lua

```diff
@@ -0,0 +1,90 @@
+if not WeakAuras.IsLibsOK() then return end
+--- @type string, Private
+local AddonName, Private = ...
+
+local WeakAuras = WeakAuras;
+local L = WeakAuras.L;
+
+--- @class SubscribableObject
+--- @field events table<string, frame[]>
+--- @field callback table<string, fun():nil>
+--- @field ClearSubscribers fun(self: SubscribableObject)
+--- @field ClearCallbacks fun(self: SubscribableObject)
+--- @field AddSubscriber fun(self: SubscribableObject, event: string, subscriber: frame, highPriority: boolean?)
+--- @field RemoveSubscriber fun(self: SubscribableObject, event: string, subscriber: frame)
+--- @field SetOnSubscriptionStatusChanged fun(self: SubscribableObject, event: string, cb: fun())
+--- @field Notify fun(self: SubscribableObject, event: type, ...: any)
+--- @field HasSubscribers fun(self: SubscribableObject, event: string): boolean
+--- @type SubscribableObject
+local SubscribableObject =
```

Comment:
unless im mistaken, this feels like it's also some groundwork for #3875?

Reply:
Maybe, I'm not actually sure if it will be. It does tick some boxes, but also quite a few things are different from what they would need to be.